trunk/NDHMS/Routing/Reservoirs/module_reservoir_read_rfc_time_series_data.F

```diff
@@ -223,7 +223,7 @@ subroutine read_time_series_file(this, time_series_file)
 
         call read_timeslice_netcdf_integer_1D_variables(ncid, 'synthetic_values', time_series_file, this%synthetic_values)
 
-        this%synthetic_values_logical = this%synthetic_values
+        this%synthetic_values_logical = merge(.true., .false., this%synthetic_values /= 0)
```

Comment:
Did this ever work?

Reply:
I tried it with GNU Fortran and it does work -- any non-zero value is true and 0 is false. This new version not only works on Cray, but I also just think it's much more explicit about what it's doing (always good).

Reply:
This one is interesting. Is it not possible to just store the mask without the merge intrinsic? I thought this is standard? It generates the mask that is then used by the merge intrinsic to set .true. and .false. to the exact values that were in the generated mask.
this%synthetic_values_logical = (this%synthetic_values .ne. 0)

Reply:
@danrosen25 I like your approach better. Same explicitness but much more concise, and many folks don't understand the merge() intrinsic. I'm going to be leaving this PR open a bit longer as although it now compiles on Cray, it crashes in some strange places due to additional Fortran-standards issues so I'll be updating this PR a bit more.